modules/java/examples/code_snippets/Examples.java

```diff
@@ -3142,30 +3142,68 @@ public void testQuerySyntaxId() throws CouchbaseLiteException {
     public void testQueryPagination() throws CouchbaseLiteException {
 
 
-        // tag::query-syntax-pagination[]
-        try {
-          this_Db = new Database("hotels");
-        } catch (CouchbaseLiteException e) {
-          e.printStackTrace();
-        }
+      // tag::query-syntax-pagination[]
+      try {
+        this_Db = new Database("hotels");
+      } catch (CouchbaseLiteException e) {
+        e.printStackTrace();
+      }
 
-        int thisOffset = 0;
-        int thisLimit = 20;
+      int thisOffset = 0;
+      int thisLimit = 20;
 
-        Query listQuery =
-                QueryBuilder
-                        .select(SelectResult.all())
-                        .from(DataSource.database(this_Db))
-                        .limit(Expression.intValue(thisLimit),
-                                  Expression.intValue(thisOffset)); // <.>
+      Query listQuery =
+      QueryBuilder
+      .select(SelectResult.all())
+      .from(DataSource.database(this_Db))
+      .limit(Expression.intValue(thisLimit),
+      Expression.intValue(thisOffset)); // <.>
 
-        // end::query-syntax-pagination[]
+      // end::query-syntax-pagination[]
 
     }
 
     // end::query-syntax-pagination-all[]
 
 
 
+    public List<Result> docsonly_QuerySyntaxN1QL (Database argDB)
+    {
+      // For Documentation -- N1QL Query using parameters
+      // tag::query-syntax-n1ql[]
+      //  Declared elsewhere: Database argDB
+
+      Database thisDb = argDB;
+
+      Query thisQuery =
+      thisDb.createQuery(
+        "SELECT META().id AS thisId FROM _ WHERE type = \"hotel\""); // <.
+
+        return thisQuery.execute().allResults();
+
+        // end::query-syntax-n1ql[]
+      }
+
+
+    public List<Result> docsonly_QuerySyntaxN1QLParams (Database argDB)
+    {
+      // For Documentation -- N1QL Query using parameters
+      // tag::query-syntax-n1ql-params[]
+      //  Declared elsewhere: Database argDB
+
+      Database thisDb = argDB;
+
+      Query thisQuery =
+          thisDb.createQuery(
+              "SELECT META().id AS thisId FROM _ WHERE type = $type"); // <.
```

Comment:
Kotlin interprets a dollar sign ("$") inside a string as a variable reference.  This code will fail to compile with a message like "undefined variable 'type'".  The dollar sign needs to be escaped:

"SELECT META().id AS thisId FROM _ WHERE type = \$type")

Reply:
This is Java @bmeike so don't need to escape it here do I? But will ensure that it is done in Kotlin module

Reply:
Duh.  This, clearly, *is* Java.  Just ignore the comment.